core/src/Kokkos_Macros.hpp

```diff
@@ -131,6 +131,11 @@
 
 #if defined(__INTEL_LLVM_COMPILER)
 #define KOKKOS_COMPILER_INTEL_LLVM __INTEL_LLVM_COMPILER
+#if defined(_MSC_VER)
+// Check this after Intel and Clang because those define _MSC_VER for
+// compatibility
+#define KOKKOS_COMPILER_MSVC _MSC_VER
+#endif
```

Comment:
Hmmmm I expect this will fail https://github.com/kokkos/kokkos/blob/b1e5aa7e9e21e5f32863e4e5311ee7f4069009e3/core/unit_test/TestCompilerMacros.cpp#L20-L29

Did you try w/o that change?

Reply:
Yeah we definitely don't want to of these defined.